src/renderer/atlas/AtlasEngine.cpp

```diff
@@ -1449,11 +1401,10 @@ void AtlasEngine::_emplaceGlyph(IDWriteFontFace* fontFace, size_t bufferPos1, si
     auto attributes = _api.attributes;
     attributes.cellCount = cellCount;
 
-    const auto [it, inserted] = _r.glyphs.emplace(std::piecewise_construct, std::forward_as_tuple(attributes, gsl::narrow<u16>(charCount), chars), std::forward_as_tuple());
-    const auto& key = it->first;
-    auto& value = it->second;
+    AtlasKey key{ attributes, gsl::narrow<u16>(charCount), chars };
+    auto valueRef = _r.glyphs.find(key);

```

Comment:
Is this a `ref` in the c++ `&` sense, or is this a ref as in "it may reference a place"

Reply:
`std::optional<std::reference_wrapper<T>>` might suffice, and make it more clear? At the same time, this might be one of the places where **not using `auto`** does the most for clarity